SimTracker/TrackTriggerAssociation/interface/TTClusterAssociationMap.h

```diff
@@ -0,0 +1,273 @@
+ *
+ */
+
+#ifndef L1_TRACK_TRIGGER_CLUSTER_ASSOCIATION_FORMAT_H
+#define L1_TRACK_TRIGGER_CLUSTER_ASSOCIATION_FORMAT_H
+
+#include "DataFormats/Common/interface/Ref.h"
+#include "DataFormats/Common/interface/Ptr.h"
+#include "DataFormats/Common/interface/DetSet.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/Common/interface/DetSetVectorNew.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/Phase2TrackerDigi/interface/Phase2TrackerDigi.h"
+#include "DataFormats/GeometryCommonDetAlgo/interface/MeasurementPoint.h"
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h" /// NOTE: this is needed even if it seems not
+#include "DataFormats/L1TrackTrigger/interface/TTCluster.h"
+#include "SimDataFormats/TrackerDigiSimLink/interface/PixelDigiSimLink.h"
+#include "SimDataFormats/Track/interface/SimTrack.h"
+#include "SimDataFormats/Track/interface/SimTrackContainer.h"
+#include "SimDataFormats/EncodedEventId/interface/EncodedEventId.h"
+#include "SimDataFormats/TrackingAnalysis/interface/TrackingParticle.h"
+
+template< typename T >
+class TTClusterAssociationMap
+{
+  public:
+    /// Constructors
+    TTClusterAssociationMap();
+
+    /// Destructor
+    ~TTClusterAssociationMap();
+
+    /// Data members:   getABC( ... )
+    /// Helper methods: findABC( ... )
+
+    /// Maps
+    std::map< edm::Ref< edmNew::DetSetVector< TTCluster< T > >, TTCluster< T > >, std::vector< edm::Ptr< TrackingParticle > > > getTTClusterToTrackingParticlesMap() const
+      { return clusterToTrackingParticleVectorMap; }
+    std::map< edm::Ptr< TrackingParticle >, std::vector< edm::Ref< edmNew::DetSetVector< TTCluster< T > >, TTCluster< T > > > > getTrackingParticleToTTClustersMap() const
+      { return trackingParticleToClusterVectorMap; }
+
+    void setTTClusterToTrackingParticlesMap( std::map< edm::Ref< edmNew::DetSetVector< TTCluster< T > >, TTCluster< T > >, std::vector< edm::Ptr< TrackingParticle > > > aMap )
+      { clusterToTrackingParticleVectorMap = aMap; }
+    void setTrackingParticleToTTClustersMap( std::map< edm::Ptr< TrackingParticle >, std::vector< edm::Ref< edmNew::DetSetVector< TTCluster< T > >, TTCluster< T > > > > aMap )
+      { trackingParticleToClusterVectorMap = aMap; }
+
+    /// Operations
+    std::vector< edm::Ptr< TrackingParticle > >                                       findTrackingParticlePtrs( edm::Ref< edmNew::DetSetVector< TTCluster< T > >, TTCluster< T > > aCluster ) const;
+    edm::Ptr< TrackingParticle >                                                      findTrackingParticlePtr( edm::Ref< edmNew::DetSetVector< TTCluster< T > >, TTCluster< T > > aCluster ) const;
+    std::vector< edm::Ref< edmNew::DetSetVector< TTCluster< T > >, TTCluster< T > > > findTTClusterRefs( edm::Ptr< TrackingParticle > aTrackingParticle ) const;
+
+    /// MC Truth methods
+    bool isGenuine( edm::Ref< edmNew::DetSetVector< TTCluster< T > >, TTCluster< T > > aCluster ) const;
```

Comment:
It would be nice if you can put in this pkg a README with the description of how the association is working, what it means Genuine, Combinatoric and Unknown, etc.. @boudoul and @delaere suggested it in the last meeting and I think it is a good idea in this case.